cypher-shell/src/integration-test/java/org/neo4j/shell/MainIntegrationTest.java

```diff
@@ -189,6 +192,33 @@ public void promptsOnPasswordChangeRequired() throws Exception {
         }
     }
 
+    @Test
+    public void allowUserToUpdateExpiredPasswordInteractivelyWithoutBeingPrompted() throws Exception {
+        //given a user that require a password change
+        shell.setCommandHelper(new CommandHelper(mock(Logger.class), Historian.empty, shell));
+        inputBuffer.put(String.format("foo%npass%nnewpass%n").getBytes());
+
+        main.connectMaybeInteractively(shell, connectionConfig, true, true );
+        String expectedLoginOutput = format( "username: neo4j%npassword: ***%n" );
+        assertEquals(expectedLoginOutput, baos.toString());
+        ensureUser();
+        int majorVersion = majorVersion(shell.getServerVersion() );
+        shell.disconnect();
+
+            //in 4.0 and later when the user attempts a non-interactive password update
+        if (majorVersion >= 4 ) {
```

Comment:
Shouldn't this be a test case of its own, instead of mixing it with the interactive test above?

Reply:
it is a test of its own, you need all that to the database in a state where there is a user that has an expired password.